compiler/rustc_session/src/parse.rs

```diff
@@ -19,6 +19,28 @@ use std::str;
 /// environment of the crate, used to drive conditional compilation.
 pub type CrateConfig = FxHashSet<(Symbol, Option<Symbol>)>;
 
+pub struct CrateCheckConfig {
+    /// Set if `names()` checking is enabled
+    pub names_checked: bool,
+    /// The union of all `names()`
+    pub names_valid: FxHashSet<Symbol>,
+    /// The set of names for which `values()` was used
+    pub values_checked: FxHashSet<Symbol>,
+    /// The set of all (name, value) pairs passed in `values()`
+    pub values_valid: FxHashSet<(Symbol, Symbol)>,
+}
+
+impl Default for CrateCheckConfig {
+    fn default() -> Self {
+        Self {
```

Comment:
```suggestion
        CheckCfg {
```
Could you avoid using `Self` outside of generic or macro context? (Here and in other places.)
It almost always makes code searchability and readability worse.